---
documentclass: book
classoption: oneside
title: "Genetic prioritisation, therapeutic repositioning and cross-disease comparisons reveal inflammatory targets tractable for kidney stone disease"
vignette: >
 %\VignetteIndexEntry{KSD Booklet}
 %\VignetteEngine{knitr::rmarkdown}
 %\VignetteEncoding{UTF-8}
output: 
  BiocStyle::html_document:
        toc: true
        toc_depth: 3
        number_sections: true
        theme: journal
        highlight: monochrome
        toc_float: true
        code_folding: hide
---

```{r, include=FALSE}
knitr::opts_chunk$set(collapse=TRUE, comment="##")
knitr::opts_chunk$set(dpi=300)
knitr::opts_chunk$set(cache=FALSE)
knitr::opts_chunk$set(echo=TRUE)
knitr::opts_chunk$set(warning=FALSE)
knitr::opts_chunk$set(message=FALSE)
knitr::opts_chunk$set(fig_cap="yes")
knitr::opts_chunk$set(eval=F)
```


# Overview {#index}

```{r logo, fig.cap="", fig.width=8.25, fig.height=4, echo=FALSE, eval=T, fig.align="left",out.width="40%"}
library(png)
library(grid)
img <- readPNG("booklet.KSD.logo.png")
grid.raster(img)
```

## Contribution

Inflammatory response plays a key role in kidney stone formation, and targeting innate immunity is an increasingly appreciated immunotherapeutic strategy in treating and preventing kidney stone disease. This study shows that identification of crosstalk between inflammatory pathways underlying disease risk can inform therapeutic targets for rational drug selection. To highlight this utility, we identify ulixertinib and losmapimod, two inhibitors of inflammatory mediators that are now under clinical investigation. An atlas of genetic targets, together with candidate drugs we reposition here, can be of translational use to manage kidney stones for years to come.


## Abstract

- **Background**: Formation of kidney stones resulting in urological disorders remains a major cause of morbidity in renal diseases and many others. Innate immunity, mainly inflammasome, has demonstrated a key role in the development of [kidney stone disease](https://www.nhs.uk/conditions/Kidney-stones/) (or `nephrolithiasis`) but a molecular rationale for therapeutic intervention targeting immunity is far from clear. We reason that identifying inflammatory gene networks underlying risk of nephrolithiasis would inform therapeutic targets for rational drug selection. 

- **Results**: Using our recently established genetics-led target prioritisation approach ([Pi](https://pubmed.ncbi.nlm.nih.gov/31253980)) applied to summary data from genome-wide association studies in nephrolithiasis (for example, [the most recent](https://pubmed.ncbi.nlm.nih.gov/31729369)), we generated a ranked list of ~12,500 target genes (see [Genetic prioritisation]), with the top ranked targets highly enriched for genes involved in the NF-kB regulation, including interaction neighbors of inflammasome genes. We identified a network of highly ranked and interconnecting genes that are of functional relevance to nephrolithiasis, such as those involved in autophagy (MTOR and SQSTM1). The identified network forms a cohesive crosstalk between inflammatory pathways (see [Pathway crosstalk]) that can be utilised for therapeutic repositioning. Coupled with removal analysis, these crosstalk genes can be also useful for combinatorial drug selection, including identification of ulixertinib (targeting MAPK3) and losmapimod (targeting MAPK14) that are both under clinical investigation as inhibitors of inflammatory mediators. Finally, we performed cross-disease comparisons and druggable pocket predictions, identifying inflammatory targets that are specific to and tractable for nephrolithiasis (see [Cross-disease comparisons]). 

- **Conclusion**: Taken together, genetic targets prioritised and candidate drugs repositioned in this study provide the rich information of how to target innate immune pathways, with the potential of advancing immunotherapeutic strategies for nephrolithiasis. The data supporting our findings are made publicly available at https://23verse.github.io/kids. Please email [us](mailto:fh12355@rjh.com.cn) for more information.


```{r, eval=FALSE, echo=FALSE}
cd ~/Sites/XGR/KSD/BOOKLET
################################################
R
library(tidyverse)

## load
load("now.KSD_booklet.RData")
## save
save(list=ls(), file="now.KSD_booklet.RData", compress="xz")

## bookdown::render_book + bookdown::gitbook
## use configuration in '_output.yml' and '_bookdown.yml'
bookdown::render_book("now.KSD_booklet.Rmd", c("bookdown::gitbook","bookdown::pdf_book","bookdown::tufte_html_book")[1])

q('no')

####################################################

##########################
https://23verse.github.io/kids
##########################
## bookdown::render_book + bookdown::gitbook
cd ~/Sites/SVN/github/kids
# delete previous *.html *.png *.css *.js *.ttf *.Rmd files *.yml files
find ./ -type f -name '*.html' -o -type f -name '*.png' -o -type f -name '*.css' -o -type f -name '*.js' -o -type f -name '*.ttf' -o -type f -name '*.Rmd' -o -type f -name '*.yml' -o -type f -name '*.RDS' -o -type f -name '*.r' | xargs rm -rf
############
# copy the whole package from ~/Sites/XGR/KSD/BOOKLET/KSD2booklet to ~/Sites/SVN/github/kids
cp -rf ~/Sites/XGR/KSD/BOOKLET/KSD2booklet/* ./
# copy/rename ~/Sites/XGR/KSD/BOOKLET/now.KSD_booklet.Rmd to ~/Sites/SVN/github/ksd/ksd.Rmd
cp -rf ~/Sites/XGR/KSD/BOOKLET/now.KSD_booklet.Rmd ./ksd.Rmd
cp -rf ~/Sites/XGR/KSD/BOOKLET/_bookdown.yml ./_bookdown.yml
cp -rf ~/Sites/XGR/KSD/BOOKLET/_output.yml ./_output.yml
# add new *.html *.png *.css *.js *.ttf *.Rmd *.yml files
git status | grep 'modified' | awk '{print $2}' | xargs git add
git status | grep 'deleted' | awk '{print $2}' | xargs git rm
find ./ -type f -name '*.html' -o -type f -name '*.png' -o -type f -name '*.css' -o -type f -name '*.js' -o -type f -name '*.ttf' -o -type f -name '*.Rmd' -o -type f -name '*.yml' -o -type f -name '*.RDS' -o -type f -name '*.r' | xargs git add
git add --all
git commit -m 'Update kids'
git push origin main
##########################

####################################################
## GitHub Pages

One site per account but unlimited projects can be created from https://pages.github.com:

> **One site `https://github.com/23verse/23verse.github.io`**

# visit https://github.com/new (enter "23verse.github.io" -> Initialize this repository with a README)
# clone into the local machine 
cd ~/Sites/SVN/github
git clone git@github.com:23verse/23verse.github.io
cd ~/Sites/SVN/github/23verse.github.io
git pull origin master
## https://github.com/23verse/23verse.github.io/settings/access: add "hfang-bristol"
echo "Hello 23verse" > index.html
git add --all
git commit -m 'Initial commit'
git push origin master
# Alive (https://23verse.github.io)

> **A project `https://github.com/23verse/kids`**

# visit https://github.com/new (enter "kids" -> Initialize this repository with a README)
## https://github.com/23verse/ksd/settings/access: add "hfang-bristol"
# clone into the local machine 
cd ~/Sites/SVN/github
git clone git@github.com:23verse/kids.git
cd ~/Sites/SVN/github/kids
git pull origin main
echo "Hello kids" > index.html
git add --all
git commit -m 'Initial commit'
git push origin main
# visit https://github.com/23verse/kids/settings -> GitHub Pages (main/root)
# Alive (https://23verse.github.io/kids): Genetic target prioritisation in nephrolithiasis
####################################################
```

```{r, eval=F, echo=F}
library(tidyverse)
RData.location <- "~/Sites/XGR/KSD/BOOKLET"
```

```{r, eval=F, echo=F}
## KSD_priority.txt.gz
cp ~/Databases/grant_app/jobs/Oxford/Oxford_Wellcome_Trust_Centre_for_Human_Genetics/ULTRA-DD/FrontierImmunology/KidneyStone/outputs/KSD_priority.txt.gz ~/Sites/XGR/KSD/BOOKLET

## subg_nodes.txt
cp ~/Databases/grant_app/jobs/Oxford/Oxford_Wellcome_Trust_Centre_for_Human_Genetics/ULTRA-DD/FrontierImmunology/KidneyStone/outputs/subg_nodes.txt ~/Sites/XGR/KSD/BOOKLET

## crossdisease_matrix.txt
cp ~/Databases/grant_app/jobs/Oxford/Oxford_Wellcome_Trust_Centre_for_Human_Genetics/ULTRA-DD/FrontierImmunology/KidneyStone/outputs/crossdisease_matrix.txt ~/Sites/XGR/KSD/BOOKLET

## crossdisease_suprahex.txt
cp ~/Databases/grant_app/jobs/Oxford/Oxford_Wellcome_Trust_Centre_for_Human_Genetics/ULTRA-DD/FrontierImmunology/KidneyStone/outputs/crossdisease_suprahex.txt ~/Sites/XGR/KSD/BOOKLET


## Crosstalk.approved_phased.txt
cp ~/Databases/grant_app/jobs/Oxford/Oxford_Wellcome_Trust_Centre_for_Human_Genetics/ULTRA-DD/FrontierImmunology/KidneyStone/Crosstalk.approved_phased.txt ~/Sites/XGR/KSD/BOOKLET
```


```{r, eval=F, echo=F}
cd ~/Databases/grant_app/jobs/Oxford/Oxford_Wellcome_Trust_Centre_for_Human_Genetics/ULTRA-DD/FrontierImmunology/KidneyStone
scp ./inputs/KSD_gwas_data.txt.gz galahad.well.ox.ac.uk:/var/www/bigdata
scp ./outputs/crossdisease_matrix.txt galahad.well.ox.ac.uk:/var/www/bigdata
```

# Genetic prioritisation

```{r manhattan, fig.cap="Manhattan plot illustrates priority rating (y-axis) for prioritised target genes (color-coded by chromosomes; x-axis), with top 30 genes named.", fig.width=10, fig.height=3.6, echo=FALSE, eval=T, fig.align="center",out.width="90%"}
library(png)
library(grid)
img <- readPNG("booklet.KSD.manhattan.png")
grid.raster(img)
```

## Package installation

```{r, eval=F, echo=T}
# if BiocManager not installed, please do so:
install.packages('BiocManager')

# then, install packages Pi and tidyverse
BiocManager::install(c('Pi','tidyverse'))
```

## Do prioritisation

```{r, eval=F, echo=T}
# load packages
library(Pi)
library(tidyverse)
RData.location <- "http://galahad.well.ox.ac.uk/bigdata"

# read GWAS summary data
data.file <- file.path(RData.location, "KSD_gwas_data.txt.gz")
data <- read_delim(data.file, delim='\t') %>% as.data.frame()

# pre-define parameters and built-in datasets
guid <- 'gskpn'
LD.customised <- xRDataLoader('GWAS_LD', guid=guid)$EUR %>% xSM2DF()
significance.threshold <- 5e-8
distance.max <- 2000
decay.kernel <- "constant"
GR.SNP <- xRDataLoader('dbSNP_GWAS', guid=guid)
GR.Gene <- xRDataLoader('UCSC_knownGene', guid=guid)
include.TAD <- "GM12878" # lymphoblast, reflective of immune-context genomic 
include.HiC <- c("Monocytes","Macrophages_M0","Macrophages_M1","Macrophages_M2","Neutrophils","Megakaryocytes","Endothelial_precursors","Erythroblasts","Fetal_thymus","Naive_CD4_T_cells","Total_CD4_T_cells","Activated_total_CD4_T_cells","Nonactivated_total_CD4_T_cells","Naive_CD8_T_cells","Total_CD8_T_cells","Naive_B_cells","Total_B_cells")
include.eQTL <- c("Blood","CD14","CD4","CD8")
network.customised <- xDefineNet(network="STRING_high", STRING.only=c("experimental_score","database_score"),  RData.location=RData.location) %>% dNetInduce(nodes_query=names(GR.Gene), knn=0, largest.comp=F)

# prepare predictors
ls_pNode_genomic <- xPierSNPsAdvABF(data, LD.customised=LD.customised, significance.threshold=significance.threshold, distance.max=distance.max, decay.kernel=decay.kernel, decay.exponent=decay.exponent, GR.SNP=GR.SNP, GR.Gene=GR.Gene, include.TAD=include.TAD, include.HiC=include.HiC, include.eQTL=include.eQTL, network.customised=network.customised, RData.location=RData.location)

# Prioritisation
ls_pNode <- Filter(Negate(is.null), ls_pNode_genomic)
dTarget <- xPierMatrix(ls_pNode, displayBy="pvalue", aggregateBy="fishers", RData.location=RData.location)

# write into a file 'KSD_priority.txt.gz'
dTarget$priority %>% select(name,rank,rating,description) %>% write_delim('KSD_priority.txt.gz', delim='\t')
```

## Prioritised genes

Target genes stored in the output file `KSD_priority.txt.gz` above can be explored below. Notes, genes are ranked by priority rating (scored 0-5; see the column `Rating`).

```{r, echo=FALSE, eval=T}
read_delim("KSD_priority.txt.gz", delim="\t") %>% transmute(Target=name, Rank=rank, Rating=signif(rating,digits=4), Description=description) %>% DT::datatable(options=list(pageLength=10,searchHighlight=T,dom='Bfrtip',buttons=c('csv','copy'), dom='t',scrollX=T), style='default', caption="", rownames=FALSE, escape=F, extensions=c('Buttons'))
```


# Pathway crosstalk

```{r crosstalk, fig.cap="Gene-centric representation of the crosstalk, with nodes for genes and edges for interactions between nodes. Node are labelled by gene symbols along with the priority information (formatted as `[rating@rank]`) and color-coded by priority rating, and interactions derived from KEGG pathway map.", fig.width=6.4, fig.height=6, echo=FALSE, eval=T, fig.align="center",out.width="60%"}
library(png)
library(grid)
img <- readPNG("booklet.KSD.crosstalk.png")
grid.raster(img)
```

## Identify crosstalk

```{r, eval=F, echo=T}
# gene interaction information merged from KEGG pathways
ls_ig <- lapply(c("KEGG_environmental","KEGG_organismal"), function(network){
	g <- xDefineNet(network=network, guid=guid)
})
ig <- xCombineNet(ls_ig)

# keep the largest component
ig2 <- dNetInduce(g=ig, nodes_query=V(ig)$name) %>% as.undirected()

# search for a subset (with a desired node number [30,35]) of the gene network 
subg <- xPierSubnet(dTarget, network.customised=ig2, subnet.size=50)

# write into a file 'subg_nodes.txt'
dTarget$priority %>% select(name,rank,rating,description) %>% semi_join(tibble(name=V(subg)$name), by='name') %>% write_delim("subg_nodes.txt", delim='\t')
```

## Crosstalk genes

Network genes identified above (`subg_nodes.txt`) can be explored:

```{r, echo=FALSE, eval=T}
read_delim("subg_nodes.txt", delim="\t") %>% transmute(Target=name, Rank=rank, Rating=signif(rating,digits=4), Description=description) %>% DT::datatable(options=list(pageLength=10,searchHighlight=T,dom='Bfrtip',buttons=c('csv','copy'), dom='t',scrollX=T), style='default', caption="", rownames=FALSE, escape=F, extensions=c('Buttons'))
```


# Cross-disease comparisons

```{r suprahex, fig.cap="Comparisons using the supra-hexagonal map. This map was learned from the prioritisation information of 53 crosstalk genes in kidney stone disease (KSD) and 8 immune-related traits. Each map illustrates a trait-specific crosstalk gene prioritisation profile. Across traits, genes with similar prioritisation patterns are mapped onto the same or nearly position in the map. The outermost frame represents the landscape for the traits analysed, from which geometric locations of traits delineate their relationships (by the similarity of prioritisation profiles between traits). AS: Ankylosing Spondylitis; CRO: Crohn's Disease; MS: Multiple Sclerosis; PSO: Psoriasis; RA: Rheumatoid Arthritis; SLE: Systemic Lupus Erythematosus; T1D: Type I Diabetes; UC: Ulcerative Colitis.", fig.width=7.3, fig.height=6, echo=FALSE, eval=T, fig.align="center",out.width="50%"}
library(png)
library(grid)
img <- readPNG("booklet.KSD.suprahex.png")
grid.raster(img)
```

## Do comparison

```{r, eval=F, echo=T}
# read cross-disease prioritisation matrix
data.file <- file.path(RData.location, "crossdisease_matrix.txt")
data <- read_delim(data.file, delim='\t') %>% column_to_rownames('target')

# analysed using the self-organising learning algorithm
sMap <- data %>% sPipeline(scaling=1)

# the resulting map partitioned into gene clusters
sBase <- sMap %>% sDmatCluster()

# write into a file 'crossdisease_suprahex.txt'
sWriteData(sMap, data, sBase, keep.data=T) %>% transmute(Target=ID, Index=str_c('H',Hexagon_index), Cluster=str_c('C',Cluster_base), KSD,AS,PSO,UC,CRO,RA,MS,SLE,T1D) %>% write_delim('crossdisease_suprahex.txt', delim='\t')
```

## Clustered genes

Clustered genes identified above (`crossdisease_suprahex.txt`) can be explored:

```{r, echo=FALSE, eval=T}
read_delim("crossdisease_suprahex.txt", delim="\t") %>% arrange(Cluster,Index,Target) %>% mutate_at(c('KSD','AS','PSO','UC','CRO','RA','MS','SLE','T1D'), 
function(x) signif(x,digits=3)) -> df

#df %>% DT::datatable(options=list(pageLength=10,searchHighlight=T,dom='Bfrtip',buttons=c('csv','copy'), dom='t',scrollX=T), style='default', caption="", rownames=FALSE, escape=F, extensions=c('Buttons'))

df %>% DT::datatable(options=list(pageLength=10,searchHighlight=T,dom='Bfrtip',buttons=c('csv','copy'), dom='t',scrollX=T,fixedColumns=list(leftColumns=3)), style='default', caption="", rownames=FALSE, escape=F, extensions=c('Buttons','FixedColumns'))
```